knora/dsplib/utils/onto_create_ontology.py

```diff
@@ -13,598 +13,618 @@
-        print(
-            f"ERROR while trying to update project '{project_shortname}' ({project_shortcode}). The error message was: {exception}")
-        exit(1)
+    project.longname = project_definition["project"]["longname"]
+    project.description = project_definition["project"].get("descriptions")
+    project.keywords = project_definition["project"].get("keywords")
+    project_remote: Project = try_network_action(
+        object=project,
+        method="update",
+        error_message_on_failure=f"WARNING: Could not update project '{project_definition['project']['shortname']}' "
+                                 f"({project_definition['project']['shortcode']})."
+    )
+    if verbose:
+        print(f"\tUpdated project '{project_definition['project']['shortname']}' ({project_definition['project']['shortcode']}).")
+    return project_remote
 
 
-def create_groups(con: Connection, groups: list[dict[str, str]], project: Project, verbose: bool) -> dict[str, Group]:
+def _create_groups(con: Connection, groups: list[dict[str, str]], project: Project) -> Tuple[dict[str, Group], bool]:
     """
-    Creates group(s) on a DSP server from a list of group definitions
+    Creates groups on a DSP server from the "groups" section of a JSON project file. If a group cannot be created, it is
+    skipped and a warning is printed, but such a group will still be part of the returned dict.
+    Returns a tuple consisting of a dict and a bool. The dict contains the groups that have successfully been created
+    (or already exist). The bool indicates if everything went smoothly during the process. If a warning or error
+    occurred, it is False.
 
     Args:
         con: connection instance to connect to the DSP server
-        groups: List of definitions of the groups (JSON) to be created
-        project: Project the group(s) should be added to
-        verbose: Prints out more information if set to True
+        groups: "groups" section of a parsed JSON project file
+        project: Project the group(s) should be added to (must exist on DSP server)
 
     Returns:
-        Dict with group names and groups
+        dict of the form ``{group name: group object}`` with the groups that have successfully been created (or already exist). Empty dict if no group was created.
+        True if everything went smoothly, False if a warning or error occurred
     """
-    new_groups: dict[str, Group] = {}
+    overall_success = True
+    current_project_groups: dict[str, Group] = {}
+    try:
+        remote_groups: list[Group] = try_network_action(
+            object=Group,
+            method="getAllGroupsForProject",
+            kwargs={"con": con, "proj_shortcode": project.shortcode},
+            error_message_on_failure="WARNING: Unable to check if group names are already existing on DSP server, "
+                                     "because it is not possible to retrieve the remote groups from DSP server."
+        )
```

Comment:
another advantage of using lambdas here, would be that you don't need the kwargs anymore:

```python
remote_groups: list[Group] = try_network_action(
    method = lambda: Group.getAllGroupsForProject(con=con, proj_shortcode=project.shortcode),
    msg = "WARNING ..."
)
```